application/src/main/java/org/togetherjava/tjbot/commands/help/HelpThreadOverviewUpdater.java

```diff
@@ -0,0 +1,149 @@
+
+        updateOverview(event.getGuild());
+    }
+
+    private void updateOverview(@NotNull Guild guild) {
+        Optional<TextChannel> maybeStagingChannel =
+                handleRequireChannel(ChannelType.STAGING, guild);
+        Optional<TextChannel> maybeOverviewChannel =
+                handleRequireChannel(ChannelType.OVERVIEW, guild);
+
+        if (maybeStagingChannel.isEmpty() || maybeOverviewChannel.isEmpty()) {
+            return;
+        }
+
+        updateOverview(maybeStagingChannel.orElseThrow(), maybeOverviewChannel.orElseThrow());
+    }
+
+    private @NotNull Optional<TextChannel> handleRequireChannel(@NotNull ChannelType channelType,
+            @NotNull Guild guild) {
+        Predicate<String> maybeChannelName = switch (channelType) {
+            case OVERVIEW -> isOverviewChannelName;
+            case STAGING -> isStagingChannelName;
+        };
+
+        Optional<TextChannel> maybeChannel = guild.getTextChannelCache()
+            .stream()
+            .filter(channel -> maybeChannelName.test(channel.getName()))
+            .findAny();
+
+        if (maybeChannel.isEmpty()) {
+            logger.warn(
+                    "Unable to update help thread overview, did not find a {} channel matching the configured pattern '{}' for guild '{}'",
+                    channelType, config.getOverviewChannelPattern(), guild.getName());
+            return Optional.empty();
+        }
+
+        return maybeChannel;
+    }
+
+    private static void updateOverview(@NotNull IThreadContainer stagingChannel,
+            @NotNull MessageChannel overviewChannel) {
+        List<ThreadChannel> activeThreads = stagingChannel.getThreadChannels()
+            .stream()
+            .filter(Predicate.not(ThreadChannel::isArchived))
+            .toList();
+
+        // FIXME Do not add prefix if empty... (this should be reworked for nicer UI anyways)
+        StringJoiner description = new StringJoiner("\n• ", "• ", "");
+
+        for (ThreadChannel thread : activeThreads) {
+            description.add(thread.getAsMention());
+        }
+
+        MessageEmbed embed = new EmbedBuilder().setTitle("Active questions")
+            .setDescription(description.toString())
+            .setColor(AMBIENT_COLOR)
+            .build();
+
+        getLastMessage(overviewChannel).flatMap(maybeStatusMessage -> {
+            if (maybeStatusMessage.isEmpty()) {
+                return overviewChannel.sendMessageEmbeds(embed);
+            }
+
+            String statusMessageId = maybeStatusMessage.orElseThrow().getId();
+            return overviewChannel.editMessageEmbedsById(statusMessageId, embed);
+        }).queue();
+    }
+
+    private static @NotNull RestAction<Optional<Message>> getLastMessage(
+            @NotNull MessageChannel channel) {
+        return channel.getHistory().retrievePast(1).map(messages -> messages.stream().findFirst());
+    }
```

Comment:
```suggestion
    private static @NotNull RestAction<Message> getLastMessage(
            @NotNull MessageChannel channel) {
        return channel.getHistory().retrievePast(1).map(messages -> messages.stream().findFirst().getOrThrow());
    }
```
This way the error consumer can be used when there's no message, opinions?

Reply:
Yea just a suggestion, if you want to use Optional that's fine too

Reply:
actually, nevermind. we only log it anways. which we do with other failures as well. so just throwing with a nice error message is enough.

Reply:
Not sure if something changed in the meantime, but this seems to be not the case anymore.

The current call-site is:

```java
getLastMessage(overviewChannel).flatMap(maybeStatusMessage -> {
    if (maybeStatusMessage.isEmpty()) {
        return overviewChannel.sendMessageEmbeds(embed);
    }

    String statusMessageId = maybeStatusMessage.orElseThrow().getId();
    return overviewChannel.editMessageEmbedsById(statusMessageId, embed);
}).queue();
```

So it has specific behavior for "not found", where it will send a new message instead of editing the old.